File: Components/WinForms/VB/RichEditMainDemo/Modules/SpellChecker.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Text
Imports System.Windows.Forms
Imports System.Globalization
Imports DevExpress.XtraSpellChecker
Imports DevExpress.XtraSpellChecker.Native
Imports DevExpress.XtraSpellChecker.Parser
Imports DevExpress.XtraRichEdit.SpellChecker

Namespace DevExpress.XtraRichEdit.Demos
	Partial Public Class SpellCheckerModule
		Inherits DevExpress.XtraRichEdit.Demos.TutorialControl
		Public Sub New()
			InitializeComponent()
            RtfLoadHelper.Load("SpellChecker.rtf", richEditControl)
            CType(New RichEditDemoExceptionsHandler(richEditControl), RichEditDemoExceptionsHandler).Install()
		End Sub

		Public Overrides ReadOnly Property PrintingRichEditControl() As RichEditControl
			Get
				Return richEditControl
			End Get
		End Property

		Private Sub SpellCheckerModule_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			If DesignMode Then
				Return
			End If

			SpellCheckerHelper.AddDictionaries(sharedDictionaryStorage1)

			If checkEdit1.Checked Then
				spellChecker1.SpellCheckMode = SpellCheckMode.AsYouType
			Else
				spellChecker1.SpellCheckMode = SpellCheckMode.OnDemand
			End If
		End Sub
		Protected Overrides Sub DoShow()
			MyBase.DoShow()
			richEditControl.Focus()
		End Sub
		Private Sub checkSpellingItem_ItemClick(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs) Handles checkSpellingItem.ItemClick
			spellChecker1.Check(richEditControl)
		End Sub
		Private Sub richEditControl_ReadOnlyChanged(ByVal sender As Object, ByVal e As EventArgs) Handles richEditControl.ReadOnlyChanged
			Me.checkSpellingItem.Enabled = Not richEditControl.ReadOnly
		End Sub
		Private Sub checkEdit1_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles checkEdit1.CheckedChanged
			If checkEdit1.Checked Then
				spellChecker1.SpellCheckMode = SpellCheckMode.AsYouType
			Else
				spellChecker1.SpellCheckMode = SpellCheckMode.OnDemand
			End If
		End Sub
	End Class
End Namespace

File: Components/WinForms/CS/RichEditMainDemo/Modules/SpellChecker.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Text;
using System.Windows.Forms;
using System.Globalization;
using DevExpress.XtraSpellChecker;
using DevExpress.XtraSpellChecker.Native;
using DevExpress.XtraSpellChecker.Parser;
using DevExpress.XtraRichEdit.SpellChecker;
using DevExpress.XtraBars;

namespace DevExpress.XtraRichEdit.Demos {
    public partial class SpellCheckerModule : DevExpress.XtraRichEdit.Demos.TutorialControl {
        public SpellCheckerModule() {
            InitializeComponent();
            this.ribbonControl1.SelectedPage = this.ribbonControl1.Pages["Review"];
            if (!DesignMode)
                SpellCheckerHelper.AddDictionaries(this.spellChecker1);
            this.spellChecker1.UseSharedDictionaries = false;
            OpenXmlLoadHelper.Load("HunspellDictionaries.docx", richEditControl);
            new RichEditDemoExceptionsHandler(richEditControl).Install();
        }

        public override RichEditControl PrintingRichEditControl { get { return richEditControl; } }

        protected override void DoShow() {
            base.DoShow();
            richEditControl.Focus();
        }
        private void barCheckItem1_CheckedChanged(object sender, XtraBars.ItemClickEventArgs e) {
            this.spellChecker1.SpellCheckMode = (sender as BarCheckItem).Checked ? SpellCheckMode.AsYouType : SpellCheckMode.OnDemand;
        }
    }
}
